cmd/kubectl-gadget/utils/trace.go

```diff
@@ -104,169 +134,304 @@ func deleteTraces(contextLogger *log.Entry, traceRestClient *restclient.RESTClie
+		return gadgetv1alpha1.TraceList{}, err
 	}
 
-	restConfig, err := kubeRestConfig()
+	traces := gadgetv1alpha1.TraceList{}
+
+	err = traceRestClient.
+		Get().
+		Namespace("gadget").
+		Resource("traces").
+		VersionedParams(&listTracesOptions, scheme.ParameterCodec).
+		Do(context.TODO()).
+		Into(&traces)
 	if err != nil {
-		contextLogger.Fatalf("Error while getting rest config: %s", err)
+		return gadgetv1alpha1.TraceList{}, fmt.Errorf("Error getting traces from traceID %q: %w", traceID, err)
 	}
 
-	traceConfig := *restConfig
-	traceConfig.ContentConfig.GroupVersion = &gadgetv1alpha1.GroupVersion
-	traceConfig.APIPath = "/apis"
-	traceConfig.NegotiatedSerializer = serializer.NewCodecFactory(scheme.Scheme)
-	traceConfig.UserAgent = restclient.DefaultKubernetesUserAgent()
+	if len(traces.Items) == 0 {
+		return gadgetv1alpha1.TraceList{}, fmt.Errorf("No traces found for traceID %q!", traceID)
+	}
 
-	traceRestClient, err := restclient.UnversionedRESTClientFor(&traceConfig)
+	return traces, nil
+}
+
+// SetTraceOperation set the operation of an existing trace.
+// If trace does not exist or Annotations does not contain GADGET_OPERATION key
+// an error is returned.
+func SetTraceOperation(traceID string, operation string) error {
+	traces, err := getTraceListFromID(traceID)
 	if err != nil {
-		contextLogger.Fatalf("Error while getting trace rest client: %s", err)
+		return err
 	}
 
-	nodeFound := false
-	for _, node := range nodes.Items {
-		if params.Node != "" && node.Name != params.Node {
+	for _, trace := range traces.Items {
+		if trace.Status.State == "Completed" {
+			return errors.New("Cannot change operation of Completed trace!")
+
 			continue
 		}
-		nodeFound = true
 
-		trace := &gadgetv1alpha1.Trace{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: subCommand + "-",
-				Namespace:    "gadget",
-				Annotations: map[string]string{
-					GADGET_OPERATION: "start",
-				},
-				Labels: map[string]string{
-					"trace-template-hash": traceID,
-				},
-			},
-			Spec: gadgetv1alpha1.TraceSpec{
-				Node:   node.Name,
-				Gadget: subCommand,
-				Filter: &gadgetv1alpha1.ContainerFilter{
-					Namespace:     namespace,
-					Podname:       params.Podname,
-					ContainerName: params.Containername,
-					Labels:        labelsSelector,
-				},
-				RunMode:    "Manual",
-				OutputMode: outputMode,
-			},
+		_, present := trace.ObjectMeta.Annotations[GADGET_OPERATION]
+		if !present {
+			return fmt.Errorf("Trace with ID %q does not contain key %q", traceID, GADGET_OPERATION)
 		}
 
-		err = traceRestClient.
-			Post().
-			Namespace(trace.ObjectMeta.Namespace).
-			Resource("traces").
-			Body(trace).
-			Do(context.TODO()).
-			Error()
+		trace.ObjectMeta.Annotations[GADGET_OPERATION] = operation
+
+		err = updateTraceOperation(&trace)
 		if err != nil {
-			deleteTraces(nil, traceRestClient, traceID)
-			contextLogger.Fatalf("Error creating trace on node %s: %q", node.Name, err)
+			return err
```

Comment:
Here, I think you should actually add context information to the error.
```suggestion
			return fmt.Errorf("error updating trace operation for %s: %w", traceID, err)
```